### DISCUSSION TITLE
TuringCourses/major_mandatory/ai_security/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#178] [General]
# TuringCourses/major_mandatory/ai_security/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major_mandatory/ai_security/

**Comment by MTDickens** [comment]
CS231n 已经是老黄历了，建议看 Justin Johnson 的 Deep Learning for Computer Vision（Justin 以前在 stanford，现在去了 UMich，然后就在 UMich 开了这个新课，所以说比 stanford 的更新一些）。

这就是这门课：https://web.eecs.umich.edu/~justincj/teaching/eecs498/FA2020/ 。原视频在 YouTube 上。B 站上也有搬运：https://www.bilibili.com/video/BV1zg411a7Wi/ 。

说一下几个注意事项：

1. 如果你本地没有 cuda 环境（必要条件是有 NVidia 的显卡），作业可以在 Google Colab 上完成。如果你不想翻墙/嫌 google colab 时间短，可以手工把 Jupyter Notebook 里面的 device='cuda' 改成 device='cpu'，不过速度会很慢。
2. 推荐一个 matrix calculus 的网站，在写作业的时候可以用来计算矩阵的导数：https://www.matrixcalculus.org/

最后，这门课不仅可以学到知识，更可以让你对深度学习的整个流程更加熟悉（准确来说，是对 PyTorch 的使用非常熟悉）。
